### DISCUSSION TITLE
Has scrolling the console via API functions changed?

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#15522] [General]
For a long time, SB_LINEDOWN, SB_LINEUP, SB_TOP, SB_BOTTOM, SB_PAGEUP, and SB_PAGEDOWN all worked as lParam for WM_VSCROLL sent to a console window.  Today only SB_LINEDOWN and SB_LINEUP seem to work.  I haven't scrutinized my code; it hasn't changed in many years.

Has conhost changed in this respect?  Thanks.

**Comment by DHowett** [comment]
> Has conhost changed in this respect? Thanks.

_Huh!_ Not to our knowledge! :smile:

Anything exposed by the window is, ah, "best effort" anyway.

**Comment by vefatica** [comment]
Thanks!  I didn't think that would have changed.  It's a little routine that used to work and hasn't been touched in years.  I'll have to look at it closer.
